lib/audit/test/audit.js

```diff
@@ -141,18 +150,134 @@ module.exports = {
+          au.assert(candles.length >= 7, 'Expected at least 7 1d candles')
+          au.assert(
+            new Date(candles[candles.length - 1].timestamp).getTime() ===
+              moment()
+                .utc()
+                .startOf('day')
+                .toDate()
+                .getTime(),
+            `Expected last 1d candle to be current day UTC: ${candles[candles.length - 1].timestamp}`
+          )
+          au.assert(
+            new Date(candles[candles.length - 2].timestamp).getTime() ===
+              moment()
+                .utc()
+                .subtract(1, 'day')
+                .startOf('day')
+                .toDate()
+                .getTime(),
+            `Expected next to last 1d candle to be previous day UTC: ${candles[candles.length - 2].timestamp}`
+          )
+        }
+
+        if (interval === '1h') {
+          au.assert(candles.length >= 24, 'Expected at least 24 1h candles')
+          au.assert(
+            new Date(candles[candles.length - 1].timestamp).getTime() ===
+              moment()
+                .utc()
+                .startOf('hour')
+                .toDate()
+                .getTime(),
+            `Expected last 1h candle to be current hour UTC: ${candles[candles.length - 1].timestamp}`
+          )
+          au.assert(
+            new Date(candles[candles.length - 2].timestamp).getTime() ===
+              moment()
+                .utc()
+                .subtract(1, 'hour')
+                .startOf('hour')
+                .toDate()
+                .getTime(),
+            `Expected next to last 1h candle to be previous hour UTC: ${candles[candles.length - 2].timestamp}`
+          )
+        }
+
+        if (interval === '1m') {
+          au.assert(candles.length >= 60, 'Expected at least 60 1m candles')
+        }
+
+        candles.forEach(c => {
+          au.assertTimestampProperty(c, 'timestamp')
+
+          let date = new Date(c.timestamp)
+
+          if (interval === '1d') {
+            au.assert(date.getUTCHours() === 0, 'Expected timestamp hours to aligned to candle size in UTC')
+          }
+
+          if (['1d', '1h'].includes(interval)) {
+            au.assert(date.getUTCMinutes() === 0, 'Expected timestamp minutes to aligned to candle size in UTC')
+          }
+
+          if (['1d', '1h', '1m'].includes(interval)) {
+            au.assert(date.getUTCSeconds() === 0, 'Expected timestamp seconds to aligned to candle size in UTC')
+          }
+
+          au.assert(date.getUTCMilliseconds() === 0, 'Expected timestamp milliseconds to align to candle size in UTC')
+        })
+
+        candles.forEach(c => au.assertNumericStringProperty(c, 'open'))
+        candles.forEach(c => au.assertNumericStringProperty(c, 'high'))
+        candles.forEach(c => au.assertNumericStringProperty(c, 'low'))
+        candles.forEach(c => au.assertNumericStringProperty(c, 'close'))
+        candles.forEach(c => au.assertNumericStringProperty(c, 'volume'))
```

Comment:
I think we should do an additional sanity check here:

* `low > 0`
* `low <= open`
* `low <= close`
* `low <= high`
* `high >= open`
* `high >= close`
* `high >= low`
* `volume > 0`